calico_rkt/calico_rkt.py

```diff
@@ -1,209 +1,256 @@
+    Return Endpoint object and newly allocated IP
+
+    :param hostname (str): Host for enndpoint allocation
+    :param orchestrator_id (str): Specifies orchestrator ('rkt')
+    :param container_id (str):
+    :param netns_path (str): namespace path
+    :param interface (str): iface to use
+    :param subnet (str): Subnet to allocate IP to
+    :rtype Endpoint: Endpoint created
+    """
+    # Allocate and Assign ip address through datastore_client
+    pool, ip = _assign_to_pool(subnet)
 
     # Create Endpoint object
     try:
-        ep = client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
-                                      container_id, [IPAddress(ip)])
+        ep = datastore_client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
+                                              container_id, [ip])
     except AddrFormatError:
-        print_stderr("This node is not configured for IPv%d. Unassigning IP "\
-                      "address %s then exiting."  % ip.version, ip)
-        client.unassign_address(IPNetwork(input_json['ipam']['subnet']), ip)
+        _log.error("This node is not configured for IPv%d. Unassigning IP "
+                   "address %s then exiting." % ip.version, ip)
+        datastore_client.unassign_address(pool, ip)
         sys.exit(1)
 
     # Create the veth, move into the container namespace, add the IP and
     # set up the default routes.
     ep.mac = ep.provision_veth(Namespace(netns_path), interface)
-    client.set_endpoint(ep)
+    datastore_client.set_endpoint(ep)
+    return ep
 
-    # Let the caller know what endpoint was created.
-    return ep, ip
 
-def _container_remove(hostname, orchestrator_id, container_id, client):
+def _container_remove(hostname, orchestrator_id, container_id):
     """
     Remove the indicated container on this host from Calico networking
+
+    :param hostname (str): Host for enndpoint allocation
+    :param orchestrator_id (str): Specifies orchestrator ('rkt')
+    :param container_id (str):
     """
     # Find the endpoint ID. We need this to find any ACL rules
     try:
-        endpoint = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        endpoint = datastore_client.get_endpoint(hostname=hostname,
+                                                 orchestrator_id=orchestrator_id,
+                                                 workload_id=container_id)
     except KeyError:
-        print_stderr("Container %s doesn't contain any endpoints" % container_id)
+        _log.error("Container %s doesn't contain any endpoints" % container_id)
         sys.exit(1)
 
     # Remove any IP address assignments that this endpoint has
     for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
-        assert(net.size == 1)
-        client.unassign_address(None, net.ip)
+        assert net.size == 1, "Net Size of endpoint's ip' is not 1. %s" % net
```

Comment:
This is a pretty awkward error message.

net.size returns the number of IP addresses in the IPNetwork, so this should be something like "Only 1 address is supported per endpoint, found %s"